packages/extension-app/src/background/index.ts

```diff
@@ -19,21 +21,21 @@ async function start(): Promise<void> {
 
   /* A) Use the client directly */
   // FIXME: use B
-  client.rpcSubscribe('{"method":"chain_subscribeNewHead","params":[],"id":1,"jsonrpc":"2.0"}', (r: any) =>
-    console.log('[client] New chain head: ' + r)
-  );
-  client
-    .rpcSend('{"method":"system_networkState","params":[],"id":1,"jsonrpc":"2.0"}')
-    .then((r: any) => console.log('[client] Network state: ' + r));
+  // client.rpcSubscribe('{"method":"chain_subscribeNewHead","params":[],"id":1,"jsonrpc":"2.0"}', (r: any) =>
+  //   console.log('[client] New chain head: ' + r)
+  // );
+  // client
+  //   .rpcSend('{"method":"system_networkState","params":[],"id":1,"jsonrpc":"2.0"}')
+  //   .then((r: any) => console.log('[client] Network state: ' + r));
 
   /* B) Use WasmProviderLite */
-  // const wasmProviderLite = new WasmProviderLite(client);
-  // wasmProviderLite.send('system_networkState', []).then(r => {
-  //   console.log('[WasmProviderLite] system_networkState resolved with', r);
-  // });
-  // wasmProviderLite.subscribe('n/a', 'chain_subscribeNewHead', [], (err: any, r: any) =>
-  //   console.log('[WasmProviderLite] Subscription notification : new chain head: ', r)
-  // );
+  const wasmProviderLite = new PostMessageProvider(client);
```

Comment:
There's some confusion here.

`PostMessageProvider` should only be inside `light-apps`. It should go inside the ApiContext. It's goal is to tell the api to send all requests via `window.postMessage`.

Here, in B), the only thing needed is message relaying (pseudo code)
```typescript
browser.runtime.message.addListener((message, sendResponse)=>{
  // check if message is well-formed

  // send to client
  const response client.rpcSend/rpcSubscribe(message);

  // send back response
  sendResponse(response);
})
```

Reply:
Ah okay, thanks for the clarification.

So to make sure I understand:
`@polkadot/extension`
* listens for RPC responses from the wasm code running in the background
* posts messages to the browser window

* listens for messages posted to browser window
* forwards those messages to an RPC call

`@substrate/light-apps`
* `PostMessageProvider` imitates a WebSockets connection for `@polkadot/api` by sending and polling for messages to/from the browser `window`

Reply:
> @substrate/light-apps PostMessageProvider imitates a WebSockets connection for @polkadot/api by sending and polling for messages to/from the browser window

Yes, correct!

> @polkadot/extension

Yes, correct too. This guy just relays messages. Not sure he'll be called `@polkadot/extension`, for now i think he doesn't need to have a name since he's probably not doing much

Reply:
> provider={new PostMessageProvider(sendMessage, subscriptionHandler)}

I don't think PostMessageProvider needs any constructor params. The sendMessage and subscriptionHandler should be inside the provider itself.

Another note: https://github.com/paritytech/substrate-light-ui/pull/706#discussion_r364868607. I can have a look on how we could detect browser vs electron vs extension (probably via user-agent or something)